src/fides/api/alembic/migrations/versions/14acee6f5459_translation_data_clear_experience_configs.py

```diff
@@ -0,0 +1,565 @@
+
+        i.e., looks at all `privacy_notices` linked to the experience config, and creates a corresponding
+        experience notice record for each linked notice.
+        """
+        link_notice_query = text(
+            """
+                INSERT INTO experiencenotices (id, notice_id, experience_config_id)
+                VALUES (:record_id, :notice_id, :experience_config_id)
+                """
+        )
+        for notice_id in experience_config["privacy_notices"]:
+            bind.execute(
+                link_notice_query,
+                {
+                    "record_id": generate_record_id("exp"),
+                    "notice_id": notice_id,
+                    "experience_config_id": experience_config["id"],
+                },
+            )
+
+    def create_experience_translations(bind):
+        """
+        Creates new experience translation records for each privacy experience config record in the db.
+
+        We simply take the text that's been temporarily saved on the config record directly and move it
+        over to the translation record. The translation is assumed to be english.
+
+        This requires the privacy experience config records to have been created before this is invoked!
+        """
+        get_experience_config_ids = text(
+            """SELECT id FROM privacyexperienceconfig;
+         """
+        )
+
+        create_translation_query = text(
+            """
+            INSERT INTO experiencetranslation (id, experience_config_id, created_at, updated_at, language, is_default, accept_button_label, reject_button_label, acknowledge_button_label, banner_description, banner_title, description, title,  privacy_preferences_link_label, privacy_policy_link_label, privacy_policy_url, save_button_label)
+            SELECT :record_id, id, created_at, updated_at, :language, :is_default, accept_button_label, reject_button_label, acknowledge_button_label, banner_description, banner_title, description, title,  privacy_preferences_link_label, privacy_policy_link_label, privacy_policy_url, save_button_label
+            FROM privacyexperienceconfig
+            WHERE id = :experience_config_id
+            """
+        )
+        for res in bind.execute(get_experience_config_ids):
+            # Create a translation for each Experience
+            bind.execute(
+                create_translation_query,
+                {
+                    "record_id": generate_record_id("exp"),
+                    "language": "en_us",
+                    "is_default": True,
+                    "experience_config_id": res["id"],
+                },
+            )
+
+        # Repoint the Experience Config History to the Experience Translation, and set dismissable while we're here.
+        link_config_to_translation_id = text(
+            """
+            UPDATE privacyexperienceconfighistory
+            SET translation_id = experiencetranslation.id, dismissable = dismissable
```

Comment:
It seems like there are other new fields that need to get ported back? allow_language_selection, auto detect language, name,

Reply:
ah, good catch! i guess when i was testing they were getting a server default? but does feel better to copy the right value from the template back.

Reply:
most of these would have server defaults, maybe not name though -